visualization/app/codeCharta/ui/fileChooser/fileChooser.po.ts

```diff
@@ -0,0 +1,23 @@
+import { Page } from "puppeteer"
+import { delay } from "../../../puppeteer.helper"
+
+export class FileChooserPageObject {
+	constructor(private page: Page) {}
+
+	public async openFile(path: string) {
+		const inputUploadHandle = await this.page.$("input[type=file]")
+		await inputUploadHandle.uploadFile(path)
+
+		const [fileChooser] = await Promise.all([this.page.waitForFileChooser(), this.page.click("file-chooser-directive .toolbar-button")])
+
+		await fileChooser.accept([path])
+		await delay(2000)
```

Comment:
Is the timeout actually necessary? I looked at the documentation and it's `.accept()` returns a promise but it's not clear if it actually waits for the upload to finish. Is there anything else we have to wait for or does that not work like that? Similar question about `.cancel()`.

Reply:
Removed it. Will open another PR soon removing most of the timeouts and letting e2e tests run on master only.

e2e are really powerful and beautiful to write with page objects.